lib/utils.js

```diff
@@ -81,4 +81,36 @@ class NullLogger {
 }
 utils.nullLogger = new NullLogger();
 
+/**
+ * Deep merge two objects.
+ * @param {Object} target
+ * @param {Object} ...sources
+ * @return {Object}
+ */
+utils.mergeDeep = (target, ...sources) => {
+    const isObject = (item) => {
+        return (item && typeof item === 'object' && !Array.isArray(item));
+    };
+
+    if (!sources.length || !sources || !target || !isObject(target)) {
+        return target;
+    }
+
+    sources.forEach((source) => {
+        if (isObject(target) && isObject(source)) {
```

Comment:
So, this will work only if the 'shapes' of the source and target are the same. Meaning, something like this

```
deepMerge({ a: { b: 'bla' } }, { a: 'bla' })
```

will ignore the second object silently. I believe that in our use-case, we would benefit from documenting that only objects of the same 'shape' are supported and throwing an error if that's not the case. Why? Because we (at least for now) intend to use it only on such objects, so failing fast will help us catch some bugs earlier.

Reply:
Done.